packages/core/lib/command-utils.js

```diff
@@ -201,6 +203,41 @@ const runCommand = async function (command, options) {
   return await command.run(options);
 };
 
+const processHelpInput = inputStrings => {
+  //is it wrong syntax of help?
+  if (inputStrings.length > 2 && ["help", "--help"].includes(inputStrings[1])) {
+    const help = require("./commands/help").meta;
+    console.log(
+      "Error: please use below syntax to displaying help information\n",
+      "\n               ",
+      help.help.usage
+    );
+    process.exit();
+  }
+
+  // when `truffle --help <cmd>` is used, convert inputStrings to run as `truffle help <cmd>`
+  if (inputStrings.length > 1 && inputStrings[0] === "--help") {
+    inputStrings[inputStrings.indexOf("--help")] = "help";
+  }
+
+  // when `truffle <cmd> help | --help` is used, convert inputStrings  to run as `truffle help <cmd>`
+  if (["help", "--help"].includes(inputStrings[inputStrings.length - 1])) {
```

Comment:
this doesn't really solve the general case, what if, for example, the user enters something like
```
truffle compile --help --all --anotherFlag
```?

if you are trying to handle the case mentioned in the comments you should probably just check the second element of the array rather than count from the end of the array

Reply:
Using the second element does seem easier, thank you
In this case `truffle compile --help --all --anotherFlag` would still error tho.